docs/examples/openiotsdk_examples.md

```diff
@@ -232,33 +253,143 @@ You can call the script directly yourself.
 ${MATTER_ROOT}/scripts/examples/openiotsdk_example.sh -C test <example name>
 ```
 
-Test example in specific network namespace with TAP device mode:
+Testing an example in a specific network namespace with TAP device mode:
 
 ```
 ${MATTER_ROOT}/scripts/run_in_ns.sh ARMns ${MATTER_ROOT}/scripts/examples/openiotsdk_example.sh -C test -n ARMtap <example name>
 ```
 
 ## Debugging
 
-Debugging can be started using a VS code launch task:
+Before debugging ensure the following:
+
+1. The example you wish to debug has been compiled with debug symbols enabled
+2. You have set up the test network (if required) correctly. See the 'Networking
+   setup' section above.
+
+### General instructions
 
 ```
-Run and Debug (Ctrl+Shift+D) => Debug Open IoT SDK example application => Start
-Debugging (F5) => <example name> => (GDB target address) => (network namespace) => (network interface) => <example name>
+1. Click 'Run and Debug' from the primary side menu or press (Ctrl+Shift+D)
+2. Select 'Debug Open IoT SDK example application' from the drop down list
+3. Click 'Start Debugging'(green triangle) or press (F5)
+4. Enter:
+=> <example name>
+=> (GDB target address)
+=> (network namespace)
+=> (network interface)
+=> <example name>
 ```
 
 For debugging remote targets (i.e. run in other network namespaces) you need to
-pass hostname/IP address of external GDB target that you want to connect to
-(_GDB target address_). In case of using the
+pass the hostname/IP address of the external GDB target that you want to connect
+to (_GDB target address_). In the case of using the
 [Open IoT SDK network environment](#networking-setup) the GDB server runs inside
-a namespace and has the same IP address as bridge interface.
+a namespace and has the same IP address as the bridge interface.
 
 ```
 ${MATTER_ROOT}/scripts/run_in_ns.sh <namespace_name> ifconfig <bridge_name>
 ```
 
-**NOTE**
+**NOTES**
 
 As you can see above, you will need to select the name of the example twice.
 This is because the debug task needs to launch the run task and currently VS
 code has no way of passing parameters between tasks.
+
+There are issues with debugging the lock-app example if you happen to be using
+"--network=host" in your docker container configuration and you are trying to
+debug while connected to the VPN. The easiest solution is just to come off the
+VPN while debugging.
```

Comment:
```suggestion
debug while connected to a VPN. The easiest solution is just to come off the
VPN while debugging.
```

Reply:
I'd use an indefinite article before the first VPN as we're not referring to a specific VPN.